charmcraft/env.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Charmcraft environment utilities."""
+
+
+import os
+import sys
+
+from charmcraft.cmdbase import CommandError
+
+
+def is_charmcraft_running_from_snap():
+    """Check if charmcraft is running from the snap."""
+    return os.getenv("SNAP_NAME") == "charmcraft" and os.getenv("SNAP") is not None
+
+
+def is_charmcraft_running_in_developer_mode():
+    """Check if Charmcraft is running under developer mode."""
+    return os.getenv("CHARMCRAFT_DEVELOPER") == "1"
```

Comment:
if we want to support more "true" type values in this field, maybe
```suggestion
    return distutils.util.strtobool(os.getenv("CHARMCRAFT_DEVELOPER", "n")) == 1
```

Reply:
I considered that, but thought it'd be better to make it more restrictive (at least initially).  I'm OK either way, I'll defer to @facundobatista.

Reply:
Done